### PR TITLE
TestHDF5Deserializer: add test_string

### PR DESCRIPTION
**Pull request by boeddeker**
This PR is a suggestion. If a maintainer thinks it's faster to self-implement it, he can do this.
The `LogReport` has the function

```python
def serialize(self, serializer: HDF5Deserializer):
    # Note that this serialization may lose some information of small
    # numerical differences.
    if isinstance(serializer, serializer_module.Serializer):
        log = json.dumps(self._log)
        serializer('_log', log)
    else:
        log = serializer('_log', '')
        import pdb; pdb.set_trace()
        self._log = json.loads(log)
```

that requires to store and load strings with the serializer. Therefore there should also be a test for that.

### CODE REVIEW

File: tests/chainer_tests/serializers_tests/test_hdf5.py [hunk 1]
```diff
@@ -142,6 +142,17 @@ def test_deserialize_scalar(self):
         z = 5
         ret = self.deserializer('z', z)
         self.assertEqual(ret, 10)
+
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_hdf5.py]: Remove redundant spaces. (See result of Travis)

File: tests/chainer_tests/serializers_tests/test_hdf5.py [hunk 2]
```diff
@@ -142,6 +142,17 @@ def test_deserialize_scalar(self):
...
+    def test_string(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            data = 'abc'
+            path = os.path.join(tmp_dir, 'tmp.hdf5')
+            with h5py.File(path, 'w') as f:
+                f.create_dataset('str', data=data)
+            with h5py.File(path, 'r') as f:
+                deserializer = hdf5.HDF5Deserializer(f)
+                ret = deserializer('str', '')
+                assert ret == data, (ret, data)
```
Comment (unnonouno) [inline tests/chainer_tests/serializers_tests/test_hdf5.py]: Use `self.assertEqual`

**Comment by unnonouno** [comment]
Oh, `tempfile.TemporaryDirectory` is only supported on python3. Please use `mkdtemp` for python2.

**Comment by boeddeker** [comment]
ok, now it should work

**Comment by unnonouno** [comment]
Great! Thank you.

**Pull request merged**